src/EthersSafe.ts

```diff
@@ -0,0 +1,111 @@
+import { BigNumber, Wallet } from 'ethers'
+import { GnosisSafe } from '../typechain'
+import SafeAbi from './abis/SafeAbiV1-2-0.json'
+import Safe from './Safe'
+import { EthSignSignature, SafeSignature } from './utils/signatures'
+import { SafeTransaction } from './utils/transactions'
+
+class EthersSafe implements Safe {
+  #contract: GnosisSafe
+  #ethers: any
+  #signer: Wallet
+
+  constructor(ethers: any, signer: Wallet, safeAddress: string) {
+    this.#ethers = ethers
+    this.#signer = signer
+    this.#contract = new ethers.Contract(safeAddress, SafeAbi, signer)
+  }
+
+  address(): string {
+    return this.#contract.address
+  }
+
+  async getOwners(): Promise<string[]> {
+    return this.#contract.getOwners()
+  }
+
+  async getThreshold(): Promise<BigNumber> {
+    return this.#contract.getThreshold()
+  }
+
+  async signMessage(hash: string): Promise<SafeSignature> {
+    const address = await this.#signer.address
+    const messageArray = this.#ethers.utils.arrayify(hash)
+    const signature = await this.#signer.signMessage(messageArray)
+    return new EthSignSignature(address, signature)
+  }
+
+  async getTransactionHash(safeTransaction: SafeTransaction): Promise<string> {
+    const safeTransactionData = safeTransaction.data
+    const txHash = await this.#contract.getTransactionHash(
+      safeTransactionData.to,
+      safeTransactionData.value,
+      safeTransactionData.data,
+      safeTransactionData.operation,
+      safeTransactionData.safeTxGas,
+      safeTransactionData.baseGas,
+      safeTransactionData.gasPrice,
+      safeTransactionData.gasToken,
+      safeTransactionData.refundReceiver,
+      safeTransactionData.nonce
+    )
+    return txHash
+  }
+
+  async confirmTransaction(safeTransaction: SafeTransaction): Promise<void> {
+    const owners = await this.getOwners()
+    if (
+      owners.filter((owner: string) => owner.toLowerCase() === this.#signer.address.toLowerCase())
```

Comment:
[`Array.find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) seems to be a better fit there

also maybe it's worth to extract function for comparing strings as an utility function, most likely we'll have to use a lot throught the project